CMakeLists.txt

```diff
@@ -19,6 +19,26 @@ message(STATUS "Configuration: ${CONFIG_OPTIONS}")
 
 project(openscad)
 
+# Default to Release build
+if(NOT CMAKE_BUILD_TYPE)
+  message(STATUS "CMAKE_BUILD_TYPE not specified.  Defaulting to 'Release'")
+  message(STATUS "Usage: cmake -DCMAKE_BUILD_TYPE=[Debug|Release|RelWithDebInfo|MinSizeRel] .")
+  set(CMAKE_BUILD_TYPE Release)
+else()
+  message(STATUS "CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
+endif()
+
+# cmake default flags include "-DNDEBUG" on all non-Debug build types, and does not put "-DDEBUG" on Debug builds
+# OpenSCAD does the opposite of that, so we override those build-type-specific flags here
+set(CMAKE_C_FLAGS_DEBUG          "-g -DDEBUG" CACHE STRING "Set Debug flags" FORCE)
```

Comment:
All these settings sound a bit overkill as cmake already does this.
How about enabling `-DDEBUG` only for debug builds and let cmake handle the rest?
e.g. `add_compile_options("$<$<CONFIG:DEBUG>:-DDEBUG>")`